tensorzero-core/tests/e2e/optimization/dicl.rs

```diff
@@ -0,0 +1,138 @@
+use serde_json::json;
+use std::collections::HashMap;
+use tensorzero_core::db::clickhouse::test_helpers::get_clickhouse;
+use tensorzero_core::inference::types::{
+    ContentBlock, ContentBlockChatOutput, ModelInput, RequestMessage, ResolvedInput,
+    ResolvedInputMessage, ResolvedInputMessageContent, Role, Text,
+};
+use tensorzero_core::optimization::dicl::insert_dicl_examples_with_batching;
+use tensorzero_core::stored_inference::RenderedSample;
+use uuid::Uuid;
+
+fn create_test_rendered_sample(input: &str, output: &str) -> RenderedSample {
+    RenderedSample {
+        function_name: "test_function".to_string(),
+        input: ModelInput {
+            system: None,
+            messages: vec![RequestMessage {
+                role: Role::User,
+                content: vec![ContentBlock::Text(Text {
+                    text: input.to_string(),
+                })],
+            }],
+        },
+        stored_input: ResolvedInput {
+            system: None,
+            messages: vec![ResolvedInputMessage {
+                role: Role::User,
+                content: vec![ResolvedInputMessageContent::Text {
+                    value: json!(input),
+                }],
+            }],
+        },
+        output: Some(vec![ContentBlockChatOutput::Text(Text {
+            text: output.to_string(),
+        })]),
+        episode_id: Some(Uuid::now_v7()),
+        inference_id: Some(Uuid::now_v7()),
+        tool_params: None,
+        output_schema: None,
+        dispreferred_outputs: vec![],
+        tags: HashMap::new(),
+    }
+}
+
+#[tokio::test]
+async fn test_insert_dicl_examples_with_batching_success() {
+    let clickhouse = get_clickhouse().await;
+
+    let samples = vec![
+        create_test_rendered_sample("test input 1", "test output 1"),
+        create_test_rendered_sample("test input 2", "test output 2"),
+    ];
+    let embeddings = vec![vec![1.0, 2.0, 3.0], vec![4.0, 5.0, 6.0]];
+    let examples_with_embeddings: Vec<(RenderedSample, Vec<f64>)> =
+        samples.into_iter().zip(embeddings.into_iter()).collect();
+
+    let result = insert_dicl_examples_with_batching(
+        &clickhouse,
+        examples_with_embeddings,
+        "test_function_e2e",
+        "test_variant_e2e",
+        10, // batch_size
+    )
+    .await;
+
+    assert!(result.is_ok());
```

Comment:
we should also check here that this actually ended up writing the desired data by using the clickhouse client to query clickouse. Add some comments explaining this as well